src/mapping/config/MappingConfiguration.cpp

```diff
@@ -494,6 +494,14 @@ MappingConfiguration::ConfiguredMapping MappingConfiguration::createMapping(
                 "Mesh \"{0}\" was not found while creating a mapping. "
                 "Please correct the to=\"{0}\" attribute.",
                 toMeshName);
+
+  // Check for compatible mesh dimensions
+  PRECICE_CHECK(fromMesh->getDimensions() == toMesh->getDimensions(),
+                "Mapping between meshes of different dimensions is not allowed. "
```

Comment:
```suggestion
                "Mapping between meshes of different dimensions is not allowed yet. "
```
To show users that this would be the way to go eventually?